finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/PathMatcher.scala

```diff
@@ -82,7 +82,11 @@ object PathMatcher {
               vars + (exprSegment.substring(1, exprSegment.length - 1) -> pathSegment)
             )
           else
-            None
+            (exprSegment.split(":"), pathSegment.split(":")) match {
```

Comment:
It's a bit weird to allow captures of the form `{foo}:bar` but not any other form that includes `{foo}` as a substring such as `{foo}-bar` or `www.{foo}.com` for examples.  Could we maybe use a regex to make this a bit more general?

Reply:
Thank you for the feedback/suggestions.  I'll work on a more general solution.